lhotse/cut/multi.py

```diff
@@ -0,0 +1,1155 @@
+import logging
```

Comment:
There is a lot of duplicated code in this file, what are the key differences between Mono and Multi cuts? If there's not too many, maybe there is a way to make them share common codebase? In the current shape, the codebase will be a bit more difficult to maintain.

Reply:
My original plan was to inherit MultiCut from MonoCut and only implement the methods where they differ. The best way would probably be to inherit both from a common base class though.

Reply:
I think a common base class does make sense here (even though I tend to be cautious around object hierarchies with more than 2 levels). Not sure how to name it though. The key common property is that both Mono and Multi cuts refer to specific data rather than to an operation on other cuts (like MixedCut does). `DataCut`? `OriginalCut`? `SourceCut`? Maybe you can think of something better..

Reply:
I think `DataCut` seems most intuitive.

Reply:
Here's what I'm thinking. We can have 2 base classes: DataCut and DerivedCut. DataCut refers to cuts that point to actual data, while DerivedCut refers to an operation on other cuts. MonoCut and MultiCut will inherit from DataCut, and MixedCut will inherit from DerivedCut. I'm not sure if it's still useful to retain the original Cut, though.

Reply:
That might be too complicated, however. Perhaps best to just create a base class for Mono and Multi, for now.

Reply:
> That might be too complicated, however. Perhaps best to just create a base class for Mono and Multi, for now.

Yes, the other base class serves no real purpose. We want to have a common base class even if just for easier type annotations.

Reply:
Done.